kinoml/datasets/kinomescan/pkis2.py

```diff
@@ -0,0 +1,109 @@
+from collections import defaultdict
+
+import pandas as pd
+
+from .utils import KINOMEScanMapper
+from ..core import BaseDatasetProvider
+from ...core.protein import AminoAcidSequence
+from ...core.ligand import Ligand
+from ...core.measurements import PercentageDisplacementMeasurement
+from ...core.conditions import AssayConditions
+from ...utils import datapath, defaultdictwithargs
+
+
+class PKIS2DatasetProvider:
+
+    """
+    Loads PKIS2 dataset as providd in `Progress towards a public chemogenomic set
+    for protein kinases and a call for contributions` (DOI: 10.1371/journal.pone.0181585)
+
+    It will build a dataframe where the SMILES-representation of ligands are the index
+    and the columns are the kinase names. To map between KINOMEscan kinase names and
+    actual sequences, helper object ``kinoml.datatasets.kinomescan.utils.KINOMEScanMapper``
+    is instantiated as a class attribute.
+
+    Parameters
+    ----------
+    featurizers : list of callables, optional=None
```

Comment:
We need a strictly defined API (more strict than just "callable") for featurizers. Maybe derive these all from a `Featurizer` base class?

Did we want to separate `ProteinFeaturizer` from `LigandFeaturizer` from `ComplexFeaturizer` types?

Reply:
If `Featurizer` objects are going to follow the factory pattern, it might as well be a function. Classes can implement `.__call__` in addition to `.featurize` (which would be basically an alias) to support both modes.

I am not sure about being specific about the split. We'll need to come up with examples to see how they look.